walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,162 @@
+   * specified by {@code request}.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<Event> sortedEvents = new ArrayList<Event>(events);
+    Collections.sort(sortedEvents, Event.ORDER_BY_START_TIME);
+
+    ArrayList<ArrayList<TimeRange>> attendeeAvailabilities =
+        getAllAttendeeAvailabilities(sortedEvents, request.getAttendees());
+    ArrayList<TimeRange> availableTimes = allTimeRangesIntersection(attendeeAvailabilities);
+    availableTimes = removeTimesBelowDuration(availableTimes, request.getDuration());
+    return availableTimes;
+  }
+
+  /**
+   * Gets a list of all the available times that each attendee in attendees is available in the day.
+   *
+   * @param events {@code Event} Collection for the day.
+   * @param attendees The collection of attendees which we wish to find all the availabilities of.
+   * @return An ArrayList where each element is the list of times that a unique attendee is
+   *         available.
+   */
+  private ArrayList<ArrayList<TimeRange>> getAllAttendeeAvailabilities(
+      Collection<Event> events, Collection<String> attendees) {
+    ArrayList<ArrayList<TimeRange>> attendeeAvailabilities = new ArrayList<ArrayList<TimeRange>>();
+    for (String attendee : attendees) {
+      attendeeAvailabilities.add(getAttendeeAvailability(events, attendee));
+    }
+    return attendeeAvailabilities;
+  }
+
+  /**
+   * Gets the available times in a day that the given attendee is available.
+   *
+   * REQUIRES: The events given must be sorted by time.
+   *
+   * @param events {@code Event} Collection for the day.
+   * @param attendee Person who we wish to find the available times.
+   * @return {@code TimeRange} ArrayList where the attendee is available.
+   */
+  private ArrayList<TimeRange> getAttendeeAvailability(Collection<Event> events, String attendee) {
+    ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+
+    // Starting from the beginning of the day, add TimeRanges where the attendee is available. Skip
+    // over the events accordingly where the attendee is listed as attending.
+    int availableStart = TimeRange.START_OF_DAY;
+    for (Event event : events) {
+      if (event.getAttendees().contains(attendee)) {
+        int eventStart = event.getWhen().start();
+        int eventEnd = event.getWhen().end();
+
+        // Case where attendee has partially overlapping events:
+        //     |-----|
+        //        |-----|
+        if (event.getWhen().contains(availableStart)) {
+          availableStart = eventEnd;
+
+        // Case where attendee has completely overlapping events:
+        //     |------|
+        //       |--|
+        } else if (eventStart < availableStart) {
+          // Skip
+
+        } else {
+          availableTimes.add(TimeRange.fromStartEnd(availableStart, eventStart, false));
```

Comment:
The boolean "false" here is difficult for me to decipher. Could you clarify this variable with an inline /**/ comment?